src/App.jsx

```diff
@@ -14,70 +17,74 @@ export const goods = [
   'Garlic',
 ];
 
-export const App = () => (
-  <main className="section container">
-    <h1 className="title is-flex is-align-items-center">No goods selected</h1>
+export const App = () => {
+  const [currentGood, setCurrentGood] = useState('Jam');
 
-    <h1 className="title is-flex is-align-items-center">
-      Jam is selected
+  const handleAddSelectedGood = good => setCurrentGood(good);
+  const handleClearSelectedGood = () => setCurrentGood(null);
 
-      <button
-        data-cy="ClearButton"
-        type="button"
-        className="delete ml-3"
-      />
-    </h1>
+  return (
+    <main className="section container">
+      <h1 className="title is-flex is-align-items-center">
+        {currentGood ? (
+          <>
+            {`${currentGood} is selected`}
 
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
             <button
-              data-cy="AddButton"
+              data-cy="ClearButton"
               type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
+              className="delete ml-3"
+              onClick={handleClearSelectedGood}
+            />
+          </>
+        ) : (
+          'No goods selected'
+        )}
+      </h1>
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
+      <table className="table">
+        <tbody>
+          {goods.map((good) => {
+            const isGoodActive = good === currentGood;
 
-        <tr data-cy="Good" className="has-background-success-light">
-          <td>
-            <button
-              data-cy="RemoveButton"
-              type="button"
-              className="button is-info"
-            >
-              -
-            </button>
-          </td>
-
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Jam
-          </td>
-        </tr>
-
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
+            return (
+              <tr
+                data-cy="Good"
+                key={nanoid()}
```

Comment:
Key shouldn't be just a random value. It should be connected with the current element
https://www.developerway.com/posts/react-key-attribute
```suggestion
                key={good}
```

Reply:
That was a very useful article. Thanks